tests/virt.rs

```diff
@@ -396,3 +438,65 @@ fn collect_component_imports(component_bytes: Vec<u8>) -> Result<Vec<String>> {
 
     Ok(import_ids)
 }
+
+// TODO remove this stub once wasi:runtime/config is implemented by wasmtime
+struct StubConfig {
+    props: Vec<(String, String)>,
+}
+impl wasi::config::runtime::Host for StubConfig {
+    #[doc = " Gets a single opaque config value set at the given key if it exists"]
+    #[must_use]
+    #[allow(clippy::type_complexity, clippy::type_repetition_in_bounds)]
+    fn get<'life0, 'async_trait>(
+        &'life0 mut self,
+        key: String,
+    ) -> ::core::pin::Pin<
+        Box<
+            dyn ::core::future::Future<
+                    Output = Result<Option<String>, wasi::config::runtime::ConfigError>,
+                > + ::core::marker::Send
+                + 'async_trait,
+        >,
+    >
+    where
+        'life0: 'async_trait,
+        Self: 'async_trait,
+    {
+        let mut result = Ok(None);
+        for (k, v) in self.props.iter() {
+            if *k == key {
+                result = Ok(Some(v.clone()));
+            }
+        }
+        let result = std::future::ready(result);
+        let result = Box::new(result);
+        let result = Pin::new(result);
+
+        result
+    }
+
+    #[doc = " Gets a list of all set config data"]
+    #[must_use]
+    #[allow(clippy::type_complexity, clippy::type_repetition_in_bounds)]
+    fn get_all<'life0, 'async_trait>(
+        &'life0 mut self,
+    ) -> ::core::pin::Pin<
+        Box<
+            dyn ::core::future::Future<
+                    Output = Result<Vec<(String, String)>, wasi::config::runtime::ConfigError>,
+                > + ::core::marker::Send
+                + 'async_trait,
+        >,
+    >
+    where
+        'life0: 'async_trait,
+        Self: 'async_trait,
+    {
+        let result = Ok(self.props.clone());
+        let result = std::future::ready(result);
+        let result = Box::new(result);
+        let result = Pin::new(result);
```

Comment:
there's probably a more idiomatic way to construct the result. Async rust is a bit beyond me at the moment.